themes/bootstrap3/js/common.js

```diff
@@ -293,46 +308,34 @@ var VuFind = (function VuFind() {
    * @param {string}  property Target property ('innerHTML', 'outerHTML' or '' for no HTML update)
    */
   function setElementContents(elm, html, attrs = {}, property = 'innerHTML') {
-    // Extract any scripts from the HTML and add them separately so that they are executed properly:
-    const scripts = [];
     const tmpDiv = document.createElement('div');
-    tmpDiv.innerHTML = html;
-    tmpDiv.querySelectorAll('script').forEach((el) => {
-      const type = el.getAttribute('type');
+    const scripts = [];
+    tmpDiv.append(...stringToNodes(html));
+    // Cloning scripts wont work as they pass internal executed state.
+    tmpDiv.querySelectorAll('script').forEach(script => {
+      const type = script.getAttribute('type');
       if (!type || 'text/javascript' === type) {
-        scripts.push(el.cloneNode(true));
-        el.remove();
+        scripts.push(script.cloneNode(true));
+        script.remove();
       }
     });
-
-    let newElm = elm;
     if (property === 'innerHTML') {
-      elm.innerHTML = tmpDiv.innerHTML;
+      elm.textContent = '';
+      elm.append(...tmpDiv.childNodes);
```

Comment:
In this instance - and perhaps on line 313 as well - it may be clearer and more appropriate to use [`replaceChildren`](https://developer.mozilla.org/en-US/docs/Web/API/Element/replaceChildren).

Reply:
oh ye, it was added in 2020 i suppose, but I think it is enough for being used?

Reply:
@EreMaijala, do you have an opinion on this? I know you're most concerned about certain older Safari versions...

Reply:
I wouldn't use replaceChildren just yet.

Reply:
On a second thought, let's use it and just provide a polyfill ([example](https://gist.github.com/tomhodgins/14451f123e1878e766c9568ecce13878)) for browsers that don't support it.